src/Microsoft.Extensions.PlatformAbstractions/Native/NativeMethods.Unix.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.PlatformAbstractions.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Unix
+        {
+            public unsafe static string GetUname()
+            {
+                // Utsname shouldn't be larger than 2K
+                var buf = stackalloc byte[2048];
+
+                try
+                {
+                    if (uname((IntPtr)buf) != 0)
+                    {
+                        throw new PlatformNotSupportedException("Error reading Unix Name");
+                    }
+                    return Marshal.PtrToStringAnsi((IntPtr)buf);
+                }
+                catch (PlatformNotSupportedException)
```

Comment:
nit: this is a bit weird - catching an exception you have just throw. Can't you move throwing outside the try/catch block?